src/main/java/tutorlink/commandpackage/FindStudentCommand.java

```diff
@@ -1,42 +1,49 @@
 package tutorlink.commandpackage;
 
+import java.util.HashMap;
+import tutorlink.exceptionspackage.DuplicateMatricNumberException;
+import tutorlink.exceptionspackage.IllegalValueException;
+import tutorlink.exceptionspackage.StudentNotFoundException;
+import tutorlink.exceptionspackage.TutorLinkException;
 import tutorlink.listpackage.StudentList;
 import tutorlink.resultpackage.CommandResult;
+import tutorlink.appstatepackage.AppState;
+import java.util.ArrayList;
 
 public class FindStudentCommand extends Command {
 
-    public static final String COMMAND_WORD = "find_student";
-    public static final String FORMAT_ERROR_MESSAGE = "Error, expected format: " + COMMAND_WORD
-            + " n/NAME {AND/OR} " + "i/MATRIC NUMBER";
-    public static final int PREFIX_INDEX = 2;
-    public static final String REGEX = "^find_student (i/(A\\d{7}[A-Z])?( n/[\\w\\d]+)?|n/[\\w\\d]" +
-            "+( i/(A\\d{7}[A-Z])?)?)( )?$";
+    public final String[] ARGUMENT_PREFIXES = {"i/", "n/"};
+    public final String COMMAND_WORD = "find_student";
 
-    protected static final String ERROR_MESSAGE = "Student (%s) not found in Student List";
+    private static final String ERROR_BOTH_NULL = "Error! Both parameters passed are null!";
+    private static final String DUPLICATE_MATRIC_NUMBER = "Error! Duplicate Matric Numbers found!";
 
-    private static final String SUCCESS_MESSAGE = "Found the following students:";
-    protected String name;
-    protected String matricNumber;
+//    public final String FORMAT_ERROR_MESSAGE = "Error, expected format: " + COMMAND_WORD
+//            + " n/NAME {AND/OR} " + "i/MATRIC NUMBER";
+//    public static final int PREFIX_INDEX = 2;
+//    public static final String REGEX = "^find_student (i/(A\\d{7}[A-Z])?( n/[\\w\\d]+)?|n/[\\w\\d]" +
+//            "+( i/(A\\d{7}[A-Z])?)?)( )?$";
 
-    public FindStudentCommand(String name, String matricNumber) {
-        this.name = name;
-        this.matricNumber = matricNumber;
-    }
+//    protected static final String ERROR_MESSAGE = "Student (%s) not found in Student List";
+//
+//    private static final String SUCCESS_MESSAGE = "Found the following students:";
```

Comment:
should remove comment code for codestyle, can do it later

Reply:
Yeah we can resolve this later. Reason I don't delete it now is because some things can be used by the parser.